d9/providers/githubactions/.ci/.github/workflows/build_deploy_and_test.yml

```diff
@@ -23,14 +23,31 @@ env:
 
 
 jobs:
+  set_checkout_ref:
+    runs-on: ubuntu-latest
+    outputs:
+      checkout_ref: ${{ steps.set_checkout_ref.outputs.checkout_ref }}
+    steps:
+      - name: Set checkout reference
+        id: set_checkout_ref
+        run: |
+          if [ ${{ github.event_name }} == 'push' ]; then
+            echo "::set-output name=checkout_ref::${{ github.ref }}"
+          elif [[ ${{ github.event_name }} == 'pull_request' ]]; then
+            echo "::set-output name=checkout_ref::${{ github.event.pull_request.head.ref }}"
+          fi
+
   configure_env_vars:
     container:
       image: quay.io/pantheon-public/build-tools-ci:8.x-php7.4
       options: --user root
     runs-on: ubuntu-latest
+    needs: set_checkout_ref
     steps:
       - name: Checkout
         uses: actions/checkout@v2
+        with:
+          ref: ${{needs.set_checkout_ref.outputs.checkout_ref}}
```

Comment:
Hi @vinmassaro I found this context ${{ github.ref_name }} that apparently does a similar thing: https://docs.github.com/en/actions/learn-github-actions/contexts#github-context and I just learnt about it.

Would you mind giving it a try? (I'm thinking on just passing it to the ref param and therefore not needing the new job). I may be missing something and you may be able to catch it if that's the case

Thanks!

Reply:
Yes, let me try that out tomorrow and get back to you. Thanks for the review!

Reply:
@kporras07 Ok, I gave a`${{ github.ref_name }}` a try but for pull_request, it outputs a reference like 23/merge which caused the checkout step to fail. I got rid of the new step and was able to add a simpler replacement directly in the checkout reference: `${{ github.head_ref || github.ref_name }}`

This will output the correct branch name: github.head_ref for pull_request and github.ref_name for push. I tested on both without any issues. Please give it a try now, thanks!

Reply:
This is great. Thanks!